packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -16,21 +16,27 @@ export interface DataSourceSrv {
   get(name?: string | null, scopedVars?: ScopedVars): Promise<DataSourceApi>;
 
   /**
-   * Get all data sources
+   * Get a list of data sources
    */
-  getAll(): DataSourceInstanceSettings[];
-
-  /**
-   * Get all data sources except for internal ones that usually should not be listed like mixed data source.
-   */
-  getExternal(): DataSourceInstanceSettings[];
+  getList(filters?: GetDataSourceListFilters): DataSourceInstanceSettings[];
 
   /**
    * Get settings and plugin metadata by name or uid
    */
   getInstanceSettings(nameOrUid: string | null | undefined): DataSourceInstanceSettings | undefined;
 }
 
+/** @public */
+export interface GetDataSourceListFilters {
+  mixed?: boolean;
+  metrics?: boolean;
+  tracing?: boolean;
+  annotations?: boolean;
+  dashboard?: boolean;
+  variables?: boolean;
+  pluginId?: string;
+}
```

Comment:
@ryantxu not sure how much you care/use DataSourceSrv, this PR refactors this service quite a bit, unifies all the get data source arrays to a single getList function, and a bunch more related to data source selection. 

Leaving getAll and getExternal in the runtime code so plugins using them will still work at runtime